From e9aefec5d83bb4e4eb9310732dfa0766586f32c8 Mon Sep 17 00:00:00 2001
Message-ID: <e9aefec5d83bb4e4eb9310732dfa0766586f32c8.1745931839.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Wed, 2 Apr 2025 22:24:27 +0200
Subject: [PATCH net] net: airoha: Add missing filed to ppe_mbox_data struct

The official Airoha EN7581 firmware requires adding max_packet filed in
ppe_mbox_data struct while the unofficial one used to develop the Airoha
EN7581 flowtable support does not require this field.
This patch does not introduce any real backwards compatible issue since
EN7581 fw is not publicly available in linux-firmware or other
repositories (e.g. OpenWrt) yet and the official fw version will use this
new layout. For this reason this change needs to be backported.

Fixes: 23290c7bc190d ("net: airoha: Introduce Airoha NPU support")
Reviewed-by: Simon Horman <horms@kernel.org>
Reviewed-by: Jacob Keller <jacob.e.keller@intel.com>
Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 drivers/net/ethernet/airoha/airoha_npu.c | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

--- a/drivers/net/ethernet/airoha/airoha_npu.c
+++ b/drivers/net/ethernet/airoha/airoha_npu.c
@@ -125,21 +125,22 @@ struct ppe_mbox_data {
 			u8 wan_xsi;
 			u8 ct_joyme4;
 			u8 max_packet;
-			int ppe_type;
-			int wan_mode;
-			int wan_sel;
-		} init_info;
+			u8 rsv[3];
+			u32 ppe_type;
+			u32 wan_mode;
+			u32 wan_sel;
+		} __packed init_info;
 		struct {
-			int func_id;
+			u32 func_id;
 			u32 size;
 			u32 data;
-		} set_info;
+		} __packed set_info;
 		struct {
 			u32 npu_stats_addr;
 			u32 foe_stats_addr;
-		} stats_info;
+		} __packed stats_info;
 	};
-};
+} __packed;
 
 enum {
 	WLAN_FUNC_SET_WAIT_PCIE_ADDR,
